### PR TITLE
Concentric cylinders MPCD geometry

### PR DESCRIPTION
**Pull request by jinnyjc**
## Motivation and context

Resolves #???

## Change log

```

```

## Checklist:

- [ ] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.rst).
- [ ] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [ ] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

**Comment by jinnyjc** [comment]
@mphoward Please review the concentric cylinders geometry.

**Comment by mphoward** [comment]
I will wait to review this until you have added the parts to compile it & made sure there are no errors. Thanks!

**Comment by jinnyjc** [comment]
@mphoward Please review the revised concentric cylinders geometry.

File: hoomd/mpcd/CMakeLists.txt [hunk 1]
```diff
@@ -58,6 +58,7 @@ set(_forces
 
 set(_geometries
     ParallelPlateGeometry
+    ConcentricCylindersGeometry
```
Comment (mphoward) [inline hoomd/mpcd/CMakeLists.txt]: Please add files alphabetically! It makes it easier to ensure they are there.
```suggestion
    ConcentricCylindersGeometry
    ParallelPlateGeometry
```

File: hoomd/mpcd/ConcentricCylindersGeometry.cc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    {
+void export_ConcentricCylindersGeometry(pybind11::module& m)
+    {
+    pybind11::class_<ConcentricCylindersGeometry, std::shared_ptr<ConcentricCylindersGeometry>>(
+        m,
+        ConcentricCylindersGeometry::getName().c_str())
+        .def(pybind11::init<Scalar, Scalar, Scalar, bool>())
+        .def_property_readonly("R0", &ConcentricCylindersGeometry::getR0)
+        .def_property_readonly("R1", &ConcentricCylindersGeometry::getR1)
+        .def_property_readonly("speed", &ConcentricCylindersGeometry::getSpeed)
```
Comment (mphoward) [inline hoomd/mpcd/ConcentricCylindersGeometry.cc]: We prefer to use descriptive variable names in Python, so I would propose the following:
```suggestion
        .def_property_readonly("inner_radius", &ConcentricCylindersGeometry::getR0)
        .def_property_readonly("outer_radius", &ConcentricCylindersGeometry::getR1)
        .def_property_readonly("angular_speed", &ConcentricCylindersGeometry::getSpeed)
```
You might want to rename the getters in C++ accordingly. It is fine to use `R0`, etc. internally in the class if you prefer the shorter variable names there.

File: hoomd/mpcd/ConcentricCylindersGeometry.h [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+     *
+     * \returns True if a collision occurred, and false otherwise
+     *
+     * \post The particle position \a pos is moved to the point of reflection, the velocity \a vel
+     * is updated according to the appropriate bounce back rule, and the integration time \a dt is
+     * decreased to the amount of time remaining.
+     */
+    HOSTDEVICE bool detectCollision(Scalar3& pos, Scalar3& vel, Scalar& dt) const
+        {
+        const Scalar r = pos.x * pos.x + pos.y * pos.y;
```
Comment (mphoward) [inline hoomd/mpcd/ConcentricCylindersGeometry.h]: This is r^2, right? I would use `rsq` to make this clear.
```suggestion
        const Scalar rsq = pos.x * pos.x + pos.y * pos.y;
```

File: hoomd/mpcd/ConcentricCylindersGeometry.h [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+        /*
+         * Check if particle is in bounds
+         */
+        const signed char sign = (char)((r < m_R02) - (r > m_R12));
```
Comment (mphoward) [inline hoomd/mpcd/ConcentricCylindersGeometry.h]: Similarly here, you might find it a little easier to read the variables if you add `sq` or `_sq`
```suggestion
        const signed char sign = (char)((r < m_R0_sq) - (r > m_R1_sq));
```

File: hoomd/mpcd/ConcentricCylindersGeometry.h [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+            {
+            dt = Scalar(0);
+            return false;
+            }
+
+        /*
+         * Find the time remaining when the particle is collided with wall. This time is computed
+         * by backtracking the position, dot(pos, pos) = R^2.
+         */
+        const Scalar R = (sign == 1) ? m_R02 : m_R12;
```
Comment (mphoward) [inline hoomd/mpcd/ConcentricCylindersGeometry.h]: I think this is going to cause an error since this is actually Rsq, so you should use that below:
```suggestion
        const Scalar Rsq = (sign == 1) ? m_R02 : m_R12;
```

File: hoomd/mpcd/ConcentricCylindersGeometry.h [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+        const Scalar a = vel.x * vel.x + vel.y * vel.y;
+        const Scalar b = -Scalar(2) * (pos.x * vel.x + pos.y * vel.y);
+        const Scalar c = pos.x * pos.x + pos.y * pos.y - R * R;
```
Comment (mphoward) [inline hoomd/mpcd/ConcentricCylindersGeometry.h]:
```suggestion
        const Scalar c = pos.x * pos.x + pos.y * pos.y - Rsq;
```

File: hoomd/mpcd/ConcentricCylindersGeometry.h [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+                vel.x = -vel.x + Scalar(2) * m_V * -pos.y;
+                vel.y = -vel.y + Scalar(2) * m_V * pos.x;
+                vel.z = -vel.z;
+                }
+            else
+                {
+                vel.x -= Scalar(2) * pos.x * pos.x * vel.x / (R * R);
+                vel.y -= Scalar(2) * pos.y * pos.y * vel.y / (R * R);
+                }
+            }
```
Comment (mphoward) [inline hoomd/mpcd/ConcentricCylindersGeometry.h]: I think this is equivalent with less duplicated code, right?
```suggestion
            // no-slip requires reflection of the tangential components.
            if (m_no_slip)
                {
                vel = -vel;
                if (sign == -1)
                    {
                    vel.x += Scalar(2) * m_V * -pos.y;
                    vel.y += Scalar(2) * m_V * pos.x;
                    }
                }
            // slip requires reflection of the normal components.
            else
                {
                vel.x -= Scalar(2) * pos.x * pos.x * vel.x / (R * R);
                vel.y -= Scalar(2) * pos.y * pos.y * vel.y / (R * R);
                }
```

File: hoomd/mpcd/ConcentricCylindersGeometry.h [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+        {
+        return std::string("ConcentricCylinders");
+        }
+#endif // __HIPCC__
+
+    private:
+    const Scalar m_R0;    //!< Inner radius of the cylinder
+    const Scalar m_R02;   //!< Square of inner radius
+    const Scalar m_R1;    //!< Outer radius of the cylinder
+    const Scalar m_R12;   //!< Square of outer radius
```
Comment (mphoward) [inline hoomd/mpcd/ConcentricCylindersGeometry.h]: I think you only ever actually used the squared versions of these, so you can just save those. Have the getters return the sqrt of these.
```suggestion
    const Scalar m_R0_sq;   //!< Square of inner radius
    const Scalar m_R1_sq;   //!< Square of outer radius
```

File: hoomd/mpcd/ConcentricCylindersGeometry.h [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+    const Scalar m_V;     //!< Velocity of the wall
```
Comment (mphoward) [inline hoomd/mpcd/ConcentricCylindersGeometry.h]: You might want to give this a different name since V is usually used for linear velocity, but this is an angular speed, right?
```suggestion
    const Scalar m_V;     //!< Velocity of the wall
```

File: hoomd/mpcd/ConcentricCylindersGeometry.h [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+        return (r > m_R12 || r < m_R02);
+        }
+
+    //! Get inner radius of the cylinder
+    /*!
+     * \returns Inner radius of the cylinder
+     */
+    HOSTDEVICE Scalar getR0() const
+        {
+        return m_R0;
```
Comment (mphoward) [inline hoomd/mpcd/ConcentricCylindersGeometry.h]: See comment below about why to do this:
```suggestion
        return slow::sqrt(m_R0_sq);
```
Can do the same for R1.

File: hoomd/mpcd/module.cc [hunk 11]
```diff
@@ -124,7 +137,11 @@ namespace gpu
 /*!
  * The kernel namespace contains the kernels that do the work of a kernel driver
  * in the gpu namespace. They are not part of the public interface for the MPCD component,
- * and are subject to change without notice.
+ * and are subject to change without notice.   // sphere
+    export_BounceBackStreamingMethodSphereGeometryBlockForce(m);
+    export_BounceBackStreamingMethodSphereGeometryConstantForce(m);
+    export_BounceBackStreamingMethodSphereGeometryNoForce(m);
+    export_BounceBackStreamingMethodSphereGeometrySineForce(m);
```
Comment (mphoward) [inline hoomd/mpcd/module.cc]: Please revert this change.

File: hoomd/mpcd/module.cc [hunk 12]
```diff
@@ -34,6 +34,7 @@ void export_ManualVirtualParticleFiller(pybind11::module&);
 void export_NoForce(pybind11::module&);
 void export_ParallelPlateGeometry(pybind11::module&);
 void export_ParallelPlateGeometryFiller(pybind11::module&);
+void export_ConcentricCylindersGeometry(pybind11::module&);
```
Comment (mphoward) [inline hoomd/mpcd/module.cc]: Make sure to add the filler (here and below for the GPU), then call those exports. The filler will be automatically created using the rejection sampling method.

**Review by mphoward — CHANGES_REQUESTED** [review]
Thanks Jinny! Overall this looks good to me. I had some comments on the code. I didn't check the math in detail yet.

After you make these changes, please also implement the Python interface for the geometry and unit tests to make sure it is working correctly (both function of the geometry class, and its use as a streaming geometry).

The streaming unit tests should check that bouncing one particle off the inner or outer cylinder works, and with the outer cylinder stationary or not. They can be simpler than the ones that are already there (i.e., it's fine to just do one particle at a time & no force).

**Comment by jinnyjc** [comment]
@mphoward Please review the implemented unit tests!

File: hoomd/mpcd/ConcentricCylindersGeometry.h [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+    HOSTDEVICE bool detectCollision(Scalar3& pos, Scalar3& vel, Scalar& dt) const
+        {
+        const Scalar rsq = pos.x * pos.x + pos.y * pos.y;
+        /*
+         * Check if particle is in bounds
+         */
+        const signed char sign = (char)((rsq < m_R0_sq) - (rsq > m_R1_sq));
+        // exit immediately if no collision is found or particle is not moving normal to the wall
+        // (since no new collision could have occurred if there is no normal motion)
+        if (sign == 0 || vel.x * vel.x + vel.y * vel.y == Scalar(0))
```
Comment (mphoward) [inline hoomd/mpcd/ConcentricCylindersGeometry.h]: Since you are going to reuse this later, you should save it in a variable. Also, I would update the comment slightly because this isn't actually the normal velocity, I think?
```suggestion
        // exit immediately if no collision is found or particle is stationary in xy plane
        const Scalar vxy_sq = vel.x * vel.x + vel.y * vel.y;
        if (sign == 0 || vxy_sq == Scalar(0))
```

File: hoomd/mpcd/ConcentricCylindersGeometry.h [hunk 14]
```diff
@@ -0,0 +1,170 @@
...
+
+        /*
+         * Find the time remaining when the particle is collided with wall. This time is computed
+         * by backtracking the position, dot(pos, pos) = R^2.
+         */
+        const Scalar Rsq = (sign == 1) ? m_R0_sq : m_R1_sq;
+        const Scalar a = vel.x * vel.x + vel.y * vel.y;
+        const Scalar b = -Scalar(2) * (pos.x * vel.x + pos.y * vel.y);
+        const Scalar c = pos.x * pos.x + pos.y * pos.y - Rsq;
+        dt = (-b + sign * (slow::sqrt(b * b - Scalar(4) * a * c))) / (Scalar(2) * a);
```
Comment (mphoward) [inline hoomd/mpcd/ConcentricCylindersGeometry.h]:
```suggestion
        const Scalar b = -Scalar(2) * (pos.x * vel.x + pos.y * vel.y);
        const Scalar c = pos.x * pos.x + pos.y * pos.y - Rsq;
        dt = (-b + sign * (slow::sqrt(b * b - Scalar(4) * vxy_sq * c))) / (Scalar(2) * vxy_sq);
```

File: hoomd/mpcd/ConcentricCylindersGeometry.h [hunk 15]
```diff
@@ -0,0 +1,170 @@
...
+
+        // backtrack the particle for dt to get to point of contact
+        pos -= vel * dt;
+
+        // update velocity according to boundary conditions
+        // no-slip requires reflection of the tangential components.
+        if (m_no_slip)
+            {
+            vel = -vel;
+            if (sign == -1)
```
Comment (mphoward) [inline hoomd/mpcd/ConcentricCylindersGeometry.h:93]: Can you add a comment here about what this case covers (hitting the moving outside wall) and a short note on how the tangential components of the velocity are determined?

File: hoomd/mpcd/ConcentricCylindersGeometry.h [hunk 16]
```diff
@@ -0,0 +1,170 @@
...
+                {
+                vel.x += Scalar(2) * m_w * -pos.y;
+                vel.y += Scalar(2) * m_w * pos.x;
+                }
+            }
+        // slip requires reflection of the normal components
+        else
+            {
+            vel.x -= Scalar(2) * pos.x * pos.x * vel.x / (Rsq);
+            vel.y -= Scalar(2) * pos.y * pos.y * vel.y / (Rsq);
```
Comment (mphoward) [inline hoomd/mpcd/ConcentricCylindersGeometry.h]: A short comment here on where these formulas come from would also be helpful for future!
```suggestion
            vel.x -= Scalar(2) * pos.x * pos.x * vel.x / Rsq;
            vel.y -= Scalar(2) * pos.y * pos.y * vel.y / Rsq;
```

File: hoomd/mpcd/ConcentricCylindersGeometry.h [hunk 17]
```diff
@@ -0,0 +1,170 @@
...
+        {
+        const Scalar rsq = pos.x * pos.x + pos.y * pos.y;
+        return (rsq > m_R1_sq || rsq < m_R0_sq);
+        }
+
+    //! Get inner radius of the cylinder
+    /*!
+     * \returns Inner radius of the cylinder
+     */
+    HOSTDEVICE Scalar getR0() const
```
Comment (mphoward) [inline hoomd/mpcd/ConcentricCylindersGeometry.h]: Since this is new code, I try to name the getters to match how I will expose them to Python. This is a little more readable. It's totally fine to still use R0 internally in the class, since that is not public.
```suggestion
    HOSTDEVICE Scalar getInnerRadius() const
```

File: hoomd/mpcd/ConcentricCylindersGeometry.h [hunk 18]
```diff
@@ -0,0 +1,170 @@
...
+        {
+        return slow::sqrt(m_R0_sq);
+        }
+
+    //! Get outer radius of the cylinder
+    /*!
+     * \returns Outer radius of the cylinder
+     */
+    HOSTDEVICE Scalar getR1() const
```
Comment (mphoward) [inline hoomd/mpcd/ConcentricCylindersGeometry.h]:
```suggestion
    HOSTDEVICE Scalar getOuterRadius() const
```

File: hoomd/mpcd/geometry.py [hunk 19]
```diff
@@ -56,6 +56,77 @@ def __init__(self, no_slip):
...
+        simulation.operations.integrator.streaming_method = stream
+
+    Attributes:
+        R0 (float): Inner radius of the cylinder (*read only*).
+
+        R1 (float): Outer radius of the cylinder (*read only*).
+
+        angular_speed (float): Angular outer cylinder speed (*read only*).
+
+            `speed` will have no effect if `no_slip` is False because the slip
```
Comment (mphoward) [inline hoomd/mpcd/geometry.py]:
```suggestion
            `angular_speed` will have no effect if `no_slip` is False because the slip
```

File: hoomd/mpcd/geometry.py [hunk 20]
```diff
@@ -56,6 +56,77 @@ def __init__(self, no_slip):
         self._param_dict.update(param_dict)
 
 
+class ConcentricCylinders(Geometry):
+    r"""Concentric-cylinders channel.
+
+    Args:
+        R0 (float): Inner radius of the cylinders.
+        R1 (float): Outer radius of the cylinders
```
Comment (mphoward) [inline hoomd/mpcd/geometry.py]:
```suggestion
        inner_radius (float): Radius of inner cylinder.
        outer_radius (float): Radius of outer cylinder.
```

File: hoomd/mpcd/geometry.py [hunk 21]
```diff
@@ -56,6 +56,77 @@ def __init__(self, no_slip):
...
+        R1 (float): Outer radius of the cylinder (*read only*).
+
+        angular_speed (float): Angular outer cylinder speed (*read only*).
+
+            `speed` will have no effect if `no_slip` is False because the slip
+            surface cannot generate shear stress.
+
+    """
+
+    def __init__(self, R0, R1, angular_speed=0.0, no_slip=True):
```
Comment (mphoward) [inline hoomd/mpcd/geometry.py]:
```suggestion
    def __init__(self, inner_radius, outer_radius, angular_speed=0.0, no_slip=True):
```

File: hoomd/mpcd/geometry.py [hunk 22]
```diff
@@ -56,6 +56,77 @@ def __init__(self, no_slip):
...
+        super().__init__(no_slip)
+        param_dict = ParameterDict(
+            R0=float(R0),
+            R1=float(R1),
+            angular_speed=float(angular_speed),
+        )
```
Comment (mphoward) [inline hoomd/mpcd/geometry.py]:
```suggestion
        param_dict = ParameterDict(
            inner_radius=float(inner_radius),
            outer_radius=float(outer_radius),
            angular_speed=float(angular_speed),
        )
```

File: hoomd/mpcd/geometry.py [hunk 23]
```diff
@@ -56,6 +56,77 @@ def __init__(self, no_slip):
...
+
+        cylinders = hoomd.mpcd.geometry.ConcentricCylinders(
+            R0=2.0, R1=5.0, angular_speed=1.0, no_slip=True)
+        stream = hoomd.mpcd.stream.BounceBack(period=1, geometry=cylinders)
+        simulation.operations.integrator.streaming_method = stream
+
+    Attributes:
+        R0 (float): Inner radius of the cylinder (*read only*).
+
+        R1 (float): Outer radius of the cylinder (*read only*).
```
Comment (mphoward) [inline hoomd/mpcd/geometry.py]:
```suggestion
        inner_radius (float): Radius of inner cylinder (*read only*).

        outer_radius (float): Radius of outer cylinder (*read only*).
```

File: hoomd/mpcd/geometry.py [hunk 24]
```diff
@@ -56,6 +56,77 @@ def __init__(self, no_slip):
...
+        super().__init__(no_slip)
+        param_dict = ParameterDict(
+            R0=float(R0),
+            R1=float(R1),
+            angular_speed=float(angular_speed),
+        )
+        self._param_dict.update(param_dict)
+
+    def _attach_hook(self):
+        self._cpp_obj = _mpcd.ConcentricCylinders(self.R0, self.R1,
```
Comment (mphoward) [inline hoomd/mpcd/geometry.py]:
```suggestion
        self._cpp_obj = _mpcd.ConcentricCylinders(self.inner_radius, self.outer_radius,
```

File: hoomd/mpcd/pytest/test_geometry.py [hunk 25]
```diff
@@ -19,6 +19,48 @@ def snap():
     return snap_
 
 
+class ConcentricCylinders:
```
Comment (mphoward) [inline hoomd/mpcd/pytest/test_geometry.py:22]: Please update these tests to use `inner_radius` and `outer_radius`.

File: hoomd/mpcd/pytest/test_stream.py [hunk 26]
```diff
@@ -189,6 +189,165 @@ def test_bulk_step(self, simulation_factory, snap):
...
+        # take another step where one particle will now hit the outer wall
+        sim.run(1)
+        snap = sim.state.get_snapshot()
+        if snap.communicator.rank == 0:
+            np.testing.assert_array_almost_equal(
+                snap.mpcd.position, [[-3.9, -2.9, 0.0], [-2.05, -0.05, -0.10]])
+            np.testing.assert_array_almost_equal(
+                snap.mpcd.velocity, [[1.0, 1.0, -1.0], [1.0, 1.0, -1.0]])
+
+        # take another step, reflecting the second particle
```
Comment (mphoward) [inline hoomd/mpcd/pytest/test_stream.py]: Is this from the inner wall? If yes, update comment to make that clear

File: hoomd/mpcd/pytest/test_stream.py [hunk 27]
```diff
@@ -189,6 +189,165 @@ def test_bulk_step(self, simulation_factory, snap):
...
+        sim.run(1)
+        snap = sim.state.get_snapshot()
+        if snap.communicator.rank == 0:
+            np.testing.assert_array_almost_equal(
+                snap.mpcd.position,
+                [[-3.972, -3.028, 0.2], [-2.05, 0.05, -0.10]])
+            np.testing.assert_array_almost_equal(
+                snap.mpcd.velocity, [[0.28, -0.28, 1.0], [1.0, 1.0, -1.0]])
+
+        # take another step, reflecting perpendicular motion of second particle
```
Comment (mphoward) [inline hoomd/mpcd/pytest/test_stream.py]: Same here, check if comment should be updated

**Review by mphoward — CHANGES_REQUESTED** [review]
Thanks Jinny! This looks excellent. I had a couple small comments on the geometry implementation, and then the main thing on the Python side is to make sure you are using `inner_radius` and `outer_radius` (not `R0` and `R1`).

After you do that, there are two other things that need to be done:
* Add the geometry to `sphinx-doc/module-mpcd-geometry.rst`. This is necessary to compile the documentation.
* Add yourself to `sphinx-doc/credits.rst`. I think that list is alphabetical by first name.
* Build the sphinx documentation locally following the directions on the HOOMD documentation, and make sure that the documentation for the geometry class looks good.

Once those things are all taken care of, let's retarget this PR onto `glotzerlab/hoomd-blue` for review and merging!

**Comment by jinnyjc** [comment]
@mphoward Please review the updated Python interface!

File: hoomd/mpcd/ConcentricCylindersGeometry.h [hunk 28]
```diff
@@ -0,0 +1,179 @@
...
+            }
+
+        /*
+         * Find the time remaining when the particle is collided with wall. This time is computed
+         * by backtracking the position, dot(pos, pos) = R^2.
+         */
+        const Scalar Rsq = (sign == 1) ? m_R0_sq : m_R1_sq;
+        const Scalar b = -Scalar(2) * (pos.x * vel.x + pos.y * vel.y);
+        const Scalar c = pos.x * pos.x + pos.y * pos.y - Rsq;
+        dt = (-b + sign * (slow::sqrt(b * b - Scalar(4) * vxy_sq * c))) / (Scalar(2) * vxy_sq);
```
Comment (mphoward) [inline hoomd/mpcd/ConcentricCylindersGeometry.h]: Minor code suggestion: I try not to use more parentheses than are needed to help with readability.
```suggestion
        dt = (-b + sign * slow::sqrt(b * b - Scalar(4) * vxy_sq * c)) / (Scalar(2) * vxy_sq);
```

File: hoomd/mpcd/geometry.py [hunk 29]
```diff
@@ -56,6 +56,83 @@ def __init__(self, no_slip):
...
+        inner_radius (float): Radius of inner cylinders.
+        outer_radius (float): Radius of outer cylinders
+        angular_speed (float): Angular speed of the outer cylinder in the
+            counterclockwise direction.
+        no_slip (bool): If True, surfaces have no-slip boundary condition.
+            Otherwise, they have the slip boundary condition.
+
+    `ConcentricCylinders` confines particles between two cylinders
+    centered around the origin. The inner cylinder are stationary, but the
+    outer cylinder may put into motion with 'angular_speed'.
```
Comment (mphoward) [inline hoomd/mpcd/geometry.py]: Light edits here, and watch out that you want to use backtick not single quote to cross-reference in sphinx.
```suggestion
    centered around the origin. The inner cylinder is stationary, but the
    outer cylinder can rotate counterclockwise about the *z* axis with
    constant `angular_speed`.
```

File: sphinx-doc/module-mpcd-geometry.rst [hunk 30]
```diff
@@ -12,6 +12,7 @@ mpcd.geometry
     :nosignatures:
 
     Geometry
+    ConcentricCylinders
```
Comment (mphoward) [inline sphinx-doc/module-mpcd-geometry.rst]: HOOMD puts objects alphabetically in a module like so:
```suggestion
    ConcentricCylinders
    Geometry
```

File: sphinx-doc/module-mpcd-geometry.rst [hunk 31]
```diff
@@ -21,6 +22,7 @@ mpcd.geometry
 .. automodule:: hoomd.mpcd.geometry
     :synopsis: Geometries.
     :members: Geometry,
+              ConcentricCylinders,
```
Comment (mphoward) [inline sphinx-doc/module-mpcd-geometry.rst]: Same here
```suggestion
    :members: ConcentricCylinders,
              Geometry,
```

**Review by mphoward — CHANGES_REQUESTED** [review]
Thank you Jinny! This looks great. Can you please address the minor comments I marked, then we want to retarget this as a PR onto `glotzerlab:hoomd-blue`. Let me know if you are unsure how to do that!
